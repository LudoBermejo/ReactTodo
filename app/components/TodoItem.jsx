import React from 'react';

export default class TodoItem extends React.Component {

  render() {
    const { id, value } = this.props;
    return (
      <li id={"list_item_" + id}>
        {id}: {value}
      </li>
    );
  }

}
TodoItem.propTypes = {
  id: React.PropTypes.number.isRequired,
  value: React.PropTypes.string.isRequired
}
