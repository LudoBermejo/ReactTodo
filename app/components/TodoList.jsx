import React from 'react';
import TodoItem from 'TodoItem';

export default class TodoList extends React.Component {

  static get propTypes() {
    todoList: React.PropTypes.array.isRequired;
  }

  renderList(listOfTodos) {
    return listOfTodos.map(item => <TodoItem key={item.id} {...item} />);
  }

  render() {
    const { todoList } = this.props;
    return (
      <div>
        <h1>List of items:</h1>
        <ul>
          {this.renderList(todoList)}
        </ul>
      </div>
    );
  }

}
