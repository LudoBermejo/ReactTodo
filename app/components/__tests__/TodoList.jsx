import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TodoList from 'TodoList';
import TodoItem from 'TodoItem';
import TestUtils from 'react-addons-test-utils';

describe('TodoList', () => {
  it('renders properly', () => {
    const values = [
      {
        id: 1,
        value: 'test_one'
      },
      {
        id: 2,
        value: 'test_two'
      }
    ];

    const todoList = renderer.create(<TodoList todoList={values} />).toJSON();
    expect(todoList).toMatchSnapshot();
  });

  describe('List items properly', () => {
    it('renders all list of items', () => {
      const values = [
        {
          id: 1,
          value: 'test_one'
        },
        {
          id: 2,
          value: 'test_two'
        }
      ];

      const todoList = mount(<TodoList todoList={values} />);
      expect(todoList.find('li').length).toBe(2);
      expect(todoList.find(TodoItem).length).toBe(2);
    })
  });
});

