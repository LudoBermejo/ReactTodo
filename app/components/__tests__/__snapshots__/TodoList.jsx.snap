exports[`TodoList renders properly 1`] = `
<div>
  <h1>
    List of items:
  </h1>
  <ul>
    <li
      id="list_item_1">
      1
      : 
      test_one
    </li>
    <li
      id="list_item_2">
      2
      : 
      test_two
    </li>
  </ul>
</div>
`;
